p256k1/src/scalar.rs

```diff
@@ -120,18 +120,17 @@ impl Scalar {
             return ret;
         }
 
-        let log = n.ilog2() + 1;
-        let mut i: u32 = 0;
+        let mut i: usize = 1;
```

Comment:
@xoloki you may also define this function:

```rust
    #[inline]
    const fn log2(v: u32) -> u8 {
        31 - v.leading_zeros()
    }
```

Reply:
I like your PR, I merged it into this branch.